common/src/main/java/org/ergoplatform/appkit/Address.java

```diff
@@ -129,6 +129,21 @@ public ErgoContract toErgoContract() {
         return new ErgoTreeContract(getErgoAddress().script(), getNetworkType());
     }
 
+    /**
+     * @return true if this address is a SigmaBoolean
+     */
+    public boolean isSigmaBoolean() {
+        return isP2PK();
+    }
+
+    /**
+     * @return SigmaBoolean value of this address. Throws an error if
+     * {@link #isSigmaBoolean()} is false
+     */
+    public Values.SigmaBoolean getSigmaBoolean() {
+        return getPublicKey();
+    }
```

Comment:
It is ok for a stub. The actual implementation should check that the ErgoTree corresponding to this address is equivalent to this special case https://github.com/ScorexFoundation/sigmastate-interpreter/blob/23d4585adc999101bd393fc9c9747b1e858b0e35/sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala#L254

The logic for extracting the `prop` in the code fragment above which can be used in Appkit is simple
```scala
val prop = ergoTree.toProposition(ergoTree.isConstantSegregation)
```
Then you can check the prop is SigmaPropConstant and extract SigmaBoolean from it the same way it is done in the interpreter.

As this is very fundamental part of ErgoTree semantics I think you need to know it.

Reply:
Improved, that makes the methods more useful